diff -Naur ./asynccommand/asynccommand.vim ./asynccommand_new/asynccommand.vim
--- ./asynccommand/asynccommand.vim	Sat Mar 05 13:12:09 2011
+++ ./asynccommand_new/asynccommand.vim	Sat Mar 05 13:36:12 2011
@@ -62,12 +62,25 @@
 if has("win32")
     " Works in Windows (Win7 x64)
     function! <SID>Async_Impl(tool_cmd, vim_cmd)
-        silent exec "!start cmd /c \"".a:tool_cmd." & ".a:vim_cmd."\""
+        let l:cmd = a:tool_cmd
+
+        if !empty(a:vim_cmd)
+            let l:cmd .= " & ".a:vim_cmd
+        endif
+
+        silent exec "!start /MIN cmd /c \"".l:cmd."\""
     endfunction
 else
     " Works in linux (Ubuntu 10.04)
     function! <SID>Async_Impl(tool_cmd, vim_cmd)
-        silent exec "! (".a:tool_cmd." ; ".a:vim_cmd.") &"
+
+        let l:cmd = a:tool_cmd
+
+        if !empty(a:vim_cmd)
+            let l:cmd .= " ; ".a:vim_cmd
+        endif
+
+        silent exec "! (".l:cmd.") &"
     endfunction
 endif
 
@@ -78,7 +91,10 @@
     " Grab output and error in case there's something we should see
     let tool_cmd = a:command . printf(&shellredir, temp_file)
 
-    let vim_cmd = "vim --servername ".v:servername." --remote-expr \"" . a:vim_func . "('" . temp_file . "')\" "
+    let vim_cmd = ""
+    if !empty(a:vim_func)
+       let vim_cmd = "vim --servername ".v:servername." --remote-expr \"" . a:vim_func . "('" . temp_file . "')\" "
+    endif
 
     call <SID>Async_Impl(tool_cmd, vim_cmd)
 endfunction
